lex_node/src/lex_node.cpp

```diff
@@ -12,199 +12,68 @@
 
-  if (!request.audio_request.data.empty()) {
-    std::copy(request.audio_request.data.begin(), request.audio_request.data.end(),
-              std::ostream_iterator<unsigned char>(*io_stream));
-  } else {
-    *io_stream << request.text_request;
-  }
-  post_content_request.SetBody(io_stream);
-  AWS_LOGSTREAM_DEBUG(__func__, "PostContentRequest " << post_content_request);
-  auto post_content_result = lex_runtime_client->PostContent(post_content_request);
-  bool is_valid = true;
-  if (post_content_result.IsSuccess()) {
-    auto & result = post_content_result.GetResult();
-    AWS_LOGSTREAM_DEBUG(__func__, "PostContentResult succeeded: " << result.GetMessage());
-    // @todo: use response variable for errors.
-    /* auto error_code = */ CopyResult(result, response);
-    // if (error_code) {
-    //    is_valid = false;
-    // }
-  } else {
-    is_valid = false;
-    AWS_LOGSTREAM_ERROR(
-      __func__, "PostContentResult failed: " << post_content_result.GetError().GetMessage());
-  }
-  return is_valid;
+LexRequest& operator<<(LexRequest& out_request, const lex_common_msgs::AudioTextConversationRequest& ros_request) {
+  out_request.accept_type = ros_request.accept_type;
+  out_request.audio_request = ros_request.audio_request.data;
+  out_request.content_type = ros_request.content_type;
+  out_request.text_request = ros_request.text_request;
+  return out_request;
 }
 
-LexNode BuildLexNode(std::shared_ptr<Client::ParameterReaderInterface> params)
-{
-  LexNode lex_node;
-  if (nullptr == params) {
-    params = std::make_shared<Client::Ros1NodeParameterReader>();
-  }
-  auto lex_configuration = LoadLexParameters(*params);
-  Client::ClientConfigurationProvider configuration_provider(params);
-  auto lex_runtime_client = Aws::MakeShared<Aws::LexRuntimeService::LexRuntimeServiceClient>(
-    kAllocationTag, configuration_provider.GetClientConfiguration());
-  lex_node.ConfigureAwsLex(lex_configuration, lex_runtime_client);
-  lex_node.Init();
-  return lex_node;
+AudioTextConversation::Response& operator<<(lex_common_msgs::AudioTextConversationResponse& ros_response,
+                                            const LexResponse& lex_response) {
+  ros_response.audio_response.data = lex_response.audio_response;
+  ros_response.dialog_state = lex_response.dialog_state;
+  ros_response.intent_name = lex_response.intent_name;
+  ros_response.message_format_type = lex_response.message_format_type;
+  ros_response.text_response = lex_response.text_response;
+  ros_response.slots = std::vector<lex_common_msgs::KeyValue>();
+  std::transform(lex_response.slots.begin(), lex_response.slots.end(),
+                 std::back_inserter(ros_response.slots), [](const std::pair<std::string, std::string>& slot) {
+              lex_common_msgs::KeyValue key_value;
+              key_value.key = slot.first;
+              key_value.value = slot.second;
+              return key_value;
+          });
+  return ros_response;
```

Comment:
Is there a difference between `AudioTextConversation::Response` and `lex_common_msgs::AudioTextConversationResponse`? Be consistent by sticking to one type.

Reply:
I'm surprised this works. I thought the return value and first argument had to be the same type.

Reply:
There is no difference between the two, they are the same type. I will change it to be consistent.